File: Try_GameEngine5/src/com/example/try_gameengine/framework/IGameModel.java
package com.example.try_gameengine.framework;

import com.example.try_gameengine.Camera.Camera;
import com.example.try_gameengine.action.Time;

import android.graphics.Canvas;
import android.graphics.Paint;
import android.view.MotionEvent;
import android.view.SurfaceHolder;

public interface IGameModel {
	public Data getData();
	public void setData(Data data);
	public void registerObserver(IMoveObserver moveObserver);
	public void removeObserver(IMoveObserver moveObserver);
	public void onTouchEvent(MotionEvent event);
	public void start();
	public void setSurfaceHolder(SurfaceHolder surfaceHolder);
	public void restart();
	public void stop();
	public int getBackgroundColor();
	public void setBackgroundColor(int backgroundColor);
	public Camera getCamera();
	public void setCamera(Camera camera);
	public void addPreProcessBlock(ProcessBlock processBlock);
//	public void setTime(Time time);
//	public Time getTime();
}

File: Try_GameEngine5/src/com/example/try_gameengine/framework/IGameModel.kt
package com.example.try_gameengine.framework

import android.view.MotionEvent
import android.view.SurfaceHolder
import com.example.try_gameengine.Camera.Camera

interface IGameModel {
    fun getData(): Data?
    fun setData(data: Data?)
    fun registerObserver(moveObserver: IMoveObserver?)
    fun removeObserver(moveObserver: IMoveObserver?)
    fun onTouchEvent(event: MotionEvent?)
    fun start()
    fun setSurfaceHolder(surfaceHolder: SurfaceHolder?)
    fun restart()
    fun stop()
    fun getBackgroundColor(): Int
    fun setBackgroundColor(backgroundColor: Int)
    fun getCamera(): Camera?
    fun setCamera(camera: Camera?)
    fun addPreProcessBlock(processBlock: ProcessBlock?) //	public void setTime(Time time);
    //	public Time getTime();
}
